src/ramlev.coffee

```diff
@@ -28,15 +28,52 @@ class Ramlev
       ,
       # Parse tests from RAML
       (raml, callback) ->
-        addTests raml, tests, callback
+        addTests raml, tests, ->
+          callback(null, raml)
       ,
       # Config refaker
-      (callback) ->
+      (raml, callback) ->
+        cache = []
+
         refaker_keys = ['fakeroot', 'directory']
         refaker_opts = _.pick(config.options, refaker_keys)
-        refaker _.extend({ schemas: extractSchemas(tests) }, refaker_opts), (err, refs, schemas) ->
-          test.json = schemas[test.refaked] for test in tests when test.refaked >= 0
-          chai.tv4.addSchema(id, json) for id, json of refs
+
+        # strategy for faked-schemas
+        refaker_opts.schemas = []
```

Comment:
Could you encapsulate the shema related logic into a class and add some unit tests?